cmd/stakercli/daemon/daemoncommands.go

```diff
@@ -408,21 +390,37 @@ func stakeFromPhase1TxBTC(ctx *cli.Context) error {
 		blockHeighTxInclusion = uint64(resp.Blk.Height)
 	}
 
-	paramsForHeight := globalParams.GetVersionedGlobalParamsByHeight(blockHeighTxInclusion)
-	if paramsForHeight == nil {
-		return fmt.Errorf("error getting param version from global params %s with height %d", inputGlobalParamsFilePath, blockHeighTxInclusion)
+	btcStakingParamsForHeight := GetBtcStakingParamsForHeight(btcStakingParams.StakingParams, uint32(blockHeighTxInclusion))
```

Comment:
I think you can simplify pr greatly by using `ParamsForBTCHeight` query from Bablon node - https://github.com/babylonlabs-io/babylon/blob/main/proto/babylon/btcstaking/v1/query.proto#L37

Reply:
damn, very true 
No idea why I didn't use this in the first place, removed unnecessary code to get all the parameters